### DISCUSSION TITLE
PHP docker pathMappings doubt

### DISCUSSION THREAD

**Discussion by ghost** [mfussenegger/nvim-dap#540] [Q&A] [answered]
Hi, I am fairly new to using DAP. I was able to configure debugger locally for php but I am facing some issue with the docker setup.

xdebug is working on the php server. I am not able to correctly specify the path mapping

configuraton -  copied the path mapping section from a python configuration as I wan't able to figure out how to convert the launch.json for vscode to lua.
```
dap.adapters.php = {
  type = 'executable',
  command = 'node',
  args = { '/home/aditya/Desktop/repos/vscode-php-debug/out/phpDebug.js' }
}

dap.configurations.php = {
  {
    type = 'php',
    request = 'launch',
    name = 'Listen for Xdebug',
    port = 9003,
    pathMappings = {
        {
            localRoot = "${workspaceFolder}", -- local path
            remoteRoot = "/demo", -- container path
        },
    },
  }
}
```

Getting this when I try to set a breakpoint - `Error setting breakpoints: The "from" argument must be of type string. Received an instance of Object`

TIA :)

Edit - Able to get it running by importing the using launch.json file via `require('dap.ext.vscode').load_launchjs("/home/aditya/Desktop/launch.json", { php = {'php'} })
`.

launch.json file - 
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Listen for debugger",
            "type": "php",
            "request": "launch",
            "port": 9003,
            "pathMappings": {
                "/demo/": "${workspaceFolder}"
            }
        }
    ]
}
```

Follow up question - Any way to make this work without the launch.json file?

**Comment by mfussenegger** [comment, marked as answer]
The launch.json would translate to:

```lua
dap.configurations.php = {
  {
    type = 'php',
    request = 'launch',
    name = 'Listen for Xdebug',
    port = 9003,
    pathMappings = {
        ['/demo/'] = "${workspaceFolder}", 
    },
  }
}
```

**Reply by ghost** [reply]
Thanks a ton @mfussenegger .... It wasn't very intuitive for me... probably because I am not the familiar with lua.. can you guide me to any resource which has information about this syntax?

**Reply by mfussenegger** [reply]
See https://github.com/nanotee/nvim-lua-guide#learning-lua
